src/odemis/driver/mcs2.py

```diff
@@ -0,0 +1,999 @@
+    SA_CTL_PKEY_THD_SENSOR_MODULE_INPUT_INDEX = 0x090B00AA
+    SA_CTL_PKEY_THD_THRESHOLD_HIGH = 0x090200B4
+    SA_CTL_PKEY_THD_THRESHOLD_LOW = 0x090200B5
+    SA_CTL_PKEY_THD_INVERSION = 0x0902000E
+    # input trigger
+    SA_CTL_PKEY_DEV_INPUT_TRIG_MODE = 0x060D0087
+    SA_CTL_PKEY_DEV_INPUT_TRIG_CONDITION = 0x060D005A
+    # output trigger
+    SA_CTL_PKEY_CH_OUTPUT_TRIG_MODE = 0x060E0087
+    SA_CTL_PKEY_CH_OUTPUT_TRIG_POLARITY = 0x060E005B
+    SA_CTL_PKEY_CH_OUTPUT_TRIG_PULSE_WIDTH = 0x060E005C
+    SA_CTL_PKEY_CH_POS_COMP_START_THRESHOLD = 0x060E0058
+    SA_CTL_PKEY_CH_POS_COMP_INCREMENT = 0x060E0059
+    SA_CTL_PKEY_CH_POS_COMP_DIRECTION = 0x060E0026
+    SA_CTL_PKEY_CH_POS_COMP_LIMIT_MIN = 0x060E0020
+    SA_CTL_PKEY_CH_POS_COMP_LIMIT_MAX = 0x060E0021
+    # hand control module
+    SA_CTL_PKEY_HM_STATE = 0x020C000F
+    SA_CTL_PKEY_HM_LOCK_OPTIONS = 0x020C0083
+    SA_CTL_PKEY_HM_DEFAULT_LOCK_OPTIONS = 0x020C0084
+    # api
+    SA_CTL_PKEY_API_EVENT_NOTIFICATION_OPTIONS = 0xF010005D
+    SA_CTL_PKEY_EVENT_NOTIFICATION_OPTIONS = 0xF010005D  # deprecated
+    SA_CTL_PKEY_API_AUTO_RECONNECT = 0xF01000A1
+    SA_CTL_PKEY_AUTO_RECONNECT = 0xF01000A1  # deprecated
+    
+    # move modes
+    SA_CTL_MOVE_MODE_CL_ABSOLUTE = 0
+    SA_CTL_MOVE_MODE_CL_RELATIVE = 1
+    SA_CTL_MOVE_MODE_SCAN_ABSOLUTE = 2
+    SA_CTL_MOVE_MODE_SCAN_RELATIVE = 3
+    SA_CTL_MOVE_MODE_STEP = 4
+
+
+    def __init__(self):
+        if os.name == "nt":
+            raise NotImplemented("Windows not yet supported")
+            # WinDLL.__init__(self, "libSA_CTL.dll")  # TODO check it works
+            # atmcd64d.dll on 64 bits
+        else:
+            # Global so that its sub-libraries can access it
+            CDLL.__init__(self, "", RTLD_GLOBAL)
+
+    def __getitem__(self, name):
+        try:
+            func = super(SA_CTLDLL, self).__getitem__(name)
+        except Exception:
+            raise AttributeError("Failed to find %s" % (name,))
+        func.__name__ = name
+        func.errcheck = self.sp_errcheck
+        return func
+
+    @staticmethod
+    def sp_errcheck(result, func, args):
+        """
+        Analyse the retuhwModelrn value of a call and raise an exception in case of
+        error.
+        Follows the ctypes.errcheck callback convention
+        """
+        # everything returns DRV_SUCCESS on correct usage, _except_ GetTemperature()
+        if result != SA_CTLDLL.SA_CTL_ERROR_NONE:
+            raise SA_CTLError(result)
+
+        return result
+
+
+class SA_CTLError(Exception):
+    """
+    SA_CTL Exception
+    """
+
+    def __init__(self, error_code):
+        self.errno = error_code
+        super(SA_CTLError, self).__init__("Error %d. %s" % (error_code, SA_CTLDLL.err_code.get(error_code, "")))
+
+
+class SmarAct_MCS2(model.Actuator):
+
+    def __init__(self, name, role, locator, ref_on_init=False,axes=None, **kwargs):
+        """
+        A driver for a SmarAct MCS2 Actuator.
```

Comment:
Please describe more here what is the "MCS2" actuator. I'd say it's "the standard SmarAct motion controller".